JoinColony/colonyCDapp
# Expenditure Fund Motion

## Description

This PR wires in the logic for creating a Fund Expenditure Motion.

## Testing
- Create two colonies.
- From Colony B, pay Colony A some of Colony B's native token. 
- Unlock Colony B's native token.
- In Colony A, create an expenditure with two slots: 
   - Slot 1: Colony A's native token
   - Slot 2: Colony B's native token
  Recipient not important, can be same for both or different. Make sure you actually have the amounts needed to fund the expenditure.
- Lock the expenditure.

- Turn on the reputation monitor.
- Install governance extn.
- Ensure you have reputation in the colony, e.g. by paying yourself. 
- Click the Fund expenditure motion button. Refresh. You should see the status of the current motion. 
- Let the motion fail. Then create a new one. The failed motion should be listed as a previous motion, while the status of the new one is shown.
- Stake and finalize motion. I have not wired in the action title since it can be done by PagePro in new ui.
- Go back to expenditure and refresh. You will see it has been fully funded. 

The point of having two different tokens is to demonstrate multicall in action. The motion encodes a single "multicall" action, which itself encodes one "moveFundsBetweenPots" action per unique token address. This gets executed on finalizing the motion, and results in the expenditure being fully funded. 

**New stuff** ✨

- fundExpenditureMotion saga

**Changes** 🏗

- I have moved the estimate gas logic from the `finalizeMotion` saga into the block ingestor. This is because the multicall was failing due to an insufficient gas estimation. In the block ingestor, we know how many actions the multicall wraps, and thus can provide a more accurate gas estimation.


Resolves #1042
